### ISSUE TITLE
Segmentation Error

### ISSUE THREAD

**Issue by shahriarabdullah** [srtee/lammps-USER-CONP2#46]
I'm having segmentation error while running conp simulation for 8M steps. But no error occurs when running for only a few thousand steps. Is this normal? I'm using total 200GB memory and 5GB memory per CPU.

**Comment by srtee** [comment]
I have had successful runs of millions of time steps, but those systems might be smaller than yours. The stack trace you attached doesn't seem to lead to a useful line in the current version of the code -- can you show me what your source code has at that line in your installation?

The code around the run setup portion has always been pretty messy so I wouldn't be surprised if there are memory bugs or other kinds of bugs lurking there still. I can only advise you to write restart files pretty often and if long runs keep giving you consistent segfaults (are you issuing several short `run` commands) then we'll have more information to proceed.

**Comment by shahriarabdullah** [comment]
Below I've attached the link to the version I'm using, So the lines shown in the stack trace matches exactly that of mine -
https://github.com/srtee/lammps-USER-CONP2/blob/main/fix_conp.cpp

I'm not using the "matout" keyword as I don't need the matrix calculation. Another question I want to ask which might not be relevant to this thread, does this conp fix results the exact same output as the older conp fix (zhenxingwang)? Asking this question because the older version always calculated the matrix/inverse matrix, but in your version matrix calculation is optional.

**Comment by srtee** [comment]
Yeah, that's a very weird place for a segfault. Unless we can find out more about why this is happening, I won't be able to figure out the error.

In general this version will not produce the same results as the old CONP code. The old code was very limited and because it did not guarantee electroneutrality the results would not be fully trustworthy.
